src/target.zig

```diff
@@ -343,3 +343,79 @@ pub fn is_libcpp_lib_name(target: std.Target, name: []const u8) bool {
 pub fn hasDebugInfo(target: std.Target) bool {
     return !target.cpu.arch.isWasm();
 }
+
+pub fn fromGoTarget(go_arch: []const u8, go_os: []const u8) !std.zig.CrossTarget {
+    var os_tag: ?std.Target.Os.Tag = undefined;
+    var abi: ?std.Target.Abi = null;
+    if (std.mem.eql(u8, go_os, "native")) {
+        os_tag = null;
+    } else if (std.mem.eql(u8, go_os, "aix")) {
+        os_tag = .aix;
+    } else if (std.mem.eql(u8, go_os, "android")) {
+        os_tag = .linux;
+        abi = .android;
+    } else if (std.mem.eql(u8, go_os, "darwin")) {
+        os_tag = .macos;
+    } else if (std.mem.eql(u8, go_os, "dragonfly")) {
+        os_tag = .dragonfly;
+    } else if (std.mem.eql(u8, go_os, "freebsd")) {
+        os_tag = .freebsd;
+    } else if (std.mem.eql(u8, go_os, "illumos")) {
+        return error.UnsupportedOperatingSystem;
+    } else if (std.mem.eql(u8, go_os, "js")) {
+        return error.UnsupportedOperatingSystem;
+    } else if (std.mem.eql(u8, go_os, "linux")) {
+        os_tag = .linux;
+        abi = .musl; // go wants static linking on linux
```

Comment:
I think this is an ok default behaviour, but this should be configurable via a flag or env var.